ruby/basics/arrays.md

```diff
@@ -34,26 +49,104 @@ Array.new(3, Array.new) #=> [[], [], []]
+```
+
+The methods `#shift` and `#unshift` can also be used to operate at the beginning of an array. While `#shift` will remove the first element of an array and return that element (much like `#pop`), `#unshift` will add elements to the beginning of an array and return that array.
+
+```
+[1, 2, 3, 4].unshift(0)   #=> [0, 1, 2, 3, 4]
+
+[1, 2, 3, 4].shift        #=> [1]
+[1, 2, 3, 4].shift(2)     #=> [1, 2]
+```
+
+### Adding and subtracting arrays
+What do you think will be the outcome of `[1, 2, 3] + [3, 4, 5]`?
+
+If you guessed `[1, 2, 3, 3, 4, 5]`, congratulations! The `#+` method will return a new array built by concatenating the two arrays. The `#concat` method works the same way.
+
+```
+a = [1, 2, 3]
+b = [3, 4, 5]
+
+a + b         #=> [1, 2, 3, 3, 4, 5]
+a.concat(b)   #=> [1, 2, 3, 3, 4, 5]
+```
+
+To find the difference between two arrays, you can use the `#-` method. This method returns a copy of the original array, removing any elements that appear in the second array.
+
+```
+[1, 1, 1, 2, 2, 3, 4] - [1, 4]    #=> [2, 2, 3]
+```
+
+### Other useful methods
+As previously mentioned, there are many methods available to Ruby arrays (over 150!). Additionally, the behavior of some methods change depending on a variety of factors, such as if they take arguments or not. For at least these reasons, [ruby-doc.org](http://ruby-doc.org/) ***will*** be your best friend in maximizing your aptitude with arrays, so go soon, and go often.
+
+Nevertheless, here is a brief look at some other common methods you might run into.
+
+```
+[].empty?               #=> true
+[[]].empty?             #=> false
+[1, 2].empty?           #=> false
+
+[1, 2, 3].reverse       #=> [3, 2, 1]
+
+[1, 2, 3].include?(3)   #=> true
+[1, 2, 3].include?("3") #=> false
+
+[1, 2, 3].join          #=> "123"
+[1, 2, 3].join("-")     #=> "1-2-3"
+```
 
 ## Exercises
-A group of exercises (If Applicable) for the student to complete in relation to the topic taught in the lesson.
+* What do you think the methods `#clear`, `#insert`, `#sample`, `#shuffle`, and `#uniq` do? Look at the array class methods at ruby-doc.org [here](http://ruby-doc.org/core-2.4.0/Array.html) and look up the methods. Were you close?
+* Fork this repo and complete the exercises.
 
 ## Assignment
 * Read [Beginning Ruby](https://www.amazon.co.uk/Beginning-Ruby-Professional-Peter-Cooper/dp/1484212797/ref=sr_1_1?ie=UTF8&qid=1475608118&sr=8-1&keywords=beginning+ruby) Chapter 3: "Ruby's Building Blocks: Data, Expressions, and Flow Control", focusing on pages 47-52 (third ed.) titled "Arrays and Lists" for a preliminary treatment on what arrays can do.
 * Follow along this [article](https://launchschool.com/books/ruby/read/arrays#whatisanarray) by Launch School, and go through the exercises using IRB or any other REPL, such as [repl.it](https://repl.it/languages/ruby).
+* http://www.eriktrautman.com/posts/ruby-explained-arrays
+* http://zetcode.com/lang/rubytutorial/arrays/
```

Comment:
This is a decent resource, But I think it belongs in additional resources because of the documentation style its in.